### PR TITLE
Web user invitation model change

### PR DESCRIPTION
**Pull request by Jtang-1**
## Product Description

No visible changes

## Technical Summary

Sets up model fields for profile and location data. `Invitation` model has the existing field `supply_point` which seems to be something specific to [CommCare Supply](https://dimagi.atlassian.net/wiki/spaces/GS/pages/2144314585/Archived+-+CommCare+Supply+Proposal). But it's subsequently just used to set the Web User's location via [here](https://github.com/dimagi/commcare-hq/blob/fa37148b1bcc5525039e16ef161b56f73e2eefeb/corehq/apps/users/models.py#L530-L531) and [WebUser set_location](https://github.com/dimagi/commcare-hq/blob/fa37148b1bcc5525039e16ef161b56f73e2eefeb/corehq/apps/users/models.py#L2504-L2512) so it's not unique to supply_point. That is why I renamed it to location_id.

[Jira Ticket](https://dimagi.atlassian.net/browse/USH-4376)

## Feature Flag

No feature flag

### Safety story

testing locally, will test on staging

### Automated test coverage

tests exists that `Invitations` can be created and accepted/declined as expected.
### QA Plan

no QA

### Migrations

- [ ] The migrations in this code can be safely applied first independently of the code

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Comment by Jtang-1** [comment]
I'm a bit confused how to handle the field renaming in relation to `the migrations in this code can be safely applied first independently of the code`.

Should this PR(1) not include the migration and change code logic to support both "location_id" and "supply_point". And then in PR(2) run the migration and then PR(3) remove code that supports "supply_point" field?

File: migrations.lock [hunk 1]
```diff
@@ -1201,6 +1201,7 @@ users
  0055_add_user_data
  0056_add_manage_domain_alerts_permission
  0057_populate_sql_user_data
+ 0058_auto_20240327_0011
```
Comment (orangejenny) [inline migrations.lock:1204]: Nitpick: a human readable name would be helpful.

**Comment by orangejenny** [comment]
To make the migration independent of the code, yes you'd need multiple PRs, although I think you could do it in two. Your PR1 would also add the new column and copy over data from supply_point, then a second PR would remove the supply_point code (it might also need to run a migration to update any invitations that were created during the first deploy and therefore have a supply_point that didn't get copied over to location_id, as the migration will get run before the new code starts being used).

But...a rename in a single PR is faster and simpler than copying data between two columns. The rename will lock the table, so no data will get read/written from the wrong place, but users might encounter transient errors during deploy. I'd call invitations a secondary/tertiary workflow and I'd be inclined to go ahead with one PR and accept the risk of errors during deploy. Curious for a perspective from @esoergel or @millerdev who tend to be more methodical about migrations in deploys.

**Comment by millerdev** [comment]
I'm not familiar enough with usages of that `supply_point` column feel comfortable assessing the risk of renaming it in a single PR, but otherwise I agree with what Jenny said.

File: migrations.lock (hunk 1 above)
Comment (Jtang-1) [reply migrations.lock]: [233cf8a](https://github.com/dimagi/commcare-hq/pull/34341/commits/233cf8a9e2816d22bab3a2ea26c24fd4770cd751)

**Comment by Jtang-1** [comment]
Thanks for the explanation and both for the input. I did a quick query and in the past 2 months, there were ~100 invitations created on prod with a non-null `supply-point` so it seems like it's low risk and would prefer to take the simpler route and do it in one PR. But if reviewer thinks otherwise, I'm happy to split it up.

**Comment by orangejenny** [comment]
I forgot that the existing usage of `supply_point` is feature flagged. That makes me feel better about doing this in one PR.

**Comment by Jtang-1** [comment]
After some more thought, I'm going to go with the safer route and just split it into multiple PRs (this is the first [PR](https://github.com/dimagi/commcare-hq/pull/34359))

Also, @orangejenny from your suggestion with the second PR, would there be a transient period between the migration (to update invitations to copy supply_points to location_id) and the code switch? During the transient period, invitations would save to supply_point instead of location_id. To avoid this, in the first PR, I should add logic so new invitations save to both supply_point and location_id.

**Comment by orangejenny** [comment]
> This would be avoided if in the first PR, I added logic so new invitations save to both supply_point and location_id right?

Yes, this sounds right. FWIW, the model in my head for these kinds of migrations is [this one](https://github.com/dimagi/commcare-hq/blob/master/docs/couch_to_sql_models.rst#conceptual-steps) (not sure if those docs will be helpful since they assume the presence of syncing code, which you're not doing), where couch/sql are supply_point/location_id. But with this being small and quick, I do think you can condense it into 2 PRs instead of 3.

**Pull request closed without merging**